code/modules/mob/living/simple_mob/subtypes/animal/sif/duck.dm

```diff
@@ -57,9 +57,10 @@
 	. = ..()
 
 	var/has_food = FALSE
```

Comment:
This chunk could be condensed to `var/has_food = isliving(L) && (locate(/obj/item/weapon/reagent_containers/food) in L.get_contents()))` unless I am misunderstanding.

Reply:
More generally:
````
/mob/living/simple_mob/animal/sif/duck/IIsAlly(mob/living/L)
	. = ..() && (!istype(L) || !(locate(/obj/item/weapon/reagent_containers/food) in L.get_contents()))
````